js/csv.js

```diff
@@ -0,0 +1,230 @@
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const readCsv = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array / Object} data Data to be converted into CSV. Can be
+ * an array of strings, a JSON object, or an array of those.
+ * @param {Array} headers List of headers for the CSV file, is an array of strings.
+ * @param {String} delimiter The delimiter to be used in the CSV.
+ * @returns A string containing the information for the CSV.
+ */
+export const buildCsv = (data, headers = [], delimiter = ",") => {
+    let csv = "";
+    if (headers !== null && Object.keys(headers).length !== 0) csv += headers.toString() + "\n";
+
+    // is simple object
+    if (!Array.isArray(data)) {
+        return csv + _objectToCsv(data, headers, delimiter);
+    }
+    // is simple array
+    if (typeof data[0] === "string") return csv + _arrayToCsv(data, delimiter);
+    if (Object.keys(data).length === 0) return "";
+
+    for (let i = 0; i < Object.keys(data).length; i++) {
+        // check if array, if it is add value as is
+        if (Array.isArray(data[i])) {
+            csv += _arrayToCsv(data[i], delimiter);
+            continue;
+        }
+
+        csv += _objectToCsv(data[i], headers, delimiter);
+    }
+
+    return csv;
+};
+
+/**
+ * Converts an object into a CSV.
+ *
+ * @param {Object} data The object to be converted.
+ * @param {*} headers The headers that are going to be
+ * searched for. If not existent, use keys for current
+ * data being converted.
+ * @param {*} delimiter The delimiter for the CSV.
+ * @returns A string of the converted object.
+ */
+export const _objectToCsv = (data, headers, delimiter) => {
+    let fields = headers;
+    if (headers === null || Object.keys(headers).length === 0) fields = Object.keys(data);
+    let csv = "";
+
+    for (let j = 0; j < fields.length; j++) {
+        if (data[fields[j]] !== null) {
+            csv += _parseStringComma(data[fields[j]], delimiter) + delimiter;
+        } else csv += ",";
+    }
+    return csv.slice(0, -1) + "\n";
+};
+
+/**
+ * Converts an array into a CSV.
+ *
+ * @param {Object} data The array to be converted.
+ * @param {*} delimiter The delimiter for the CSV.
+ * @returns A string of the converted array.
+ */
+export const _arrayToCsv = (data, delimiter) => {
+    let csv = "";
+    for (let j = 0; j < data.length; j++) {
+        csv += _parseStringComma(data[j], delimiter) + delimiter;
+    }
+    return csv.slice(0, -1) + "\n";

```

Comment:
Why didnt [this](https://github.com/ripe-tech/ripe-util-vue/blob/master/mixins/csv.js#L26) work?

Reply:
It fails the tests that were created. Didn't look into it very deep, but I believe it's because of how it doesn't handle strings that include the delimiter.